tests/framework/resources.py

```diff
@@ -369,17 +369,61 @@ def put(self, **args):
         )
 
     @staticmethod
-    def create_json(mem_file_path, snapshot_path, diff=False):
+    def create_json(mem_file_path, snapshot_path, diff=False, resume=False):
         """Compose the json associated to this type of API request."""
         datax = {
             'mem_file_path': mem_file_path,
             'snapshot_path': snapshot_path,
         }
         if diff:
             datax['enable_diff_snapshots'] = True
+        if resume:
+            datax['resume_vm'] = True
         return datax
 
 
+class SnapshotHelper():
+    """Facility for creation and loading of microvm snapshots."""
+
+    def __init__(self, api_usocket_full_name, api_session):
+        """Specify the information needed for sending API requests."""
+        self._create = SnapshotCreate(api_usocket_full_name, api_session)
+        self._load = SnapshotLoad(api_usocket_full_name, api_session)
+        self._vm_state = Vm(api_usocket_full_name, api_session)
+
+    def create(self, mem_file_path, snapshot_path, diff=False, version=None):
+        """Create a snapshot of the microvm."""
+        return self._create.put(
+            mem_file_path=mem_file_path,
+            snapshot_path=snapshot_path,
+            diff=diff,
+            version=version
+        )
+
+    def load(self, mem_file_path, snapshot_path, diff=False, resume=False):
+        """Load a snapshot of the microvm."""
+        response = self._load.put(
+            mem_file_path=mem_file_path,
+            snapshot_path=snapshot_path,
+            diff=diff,
+            resume=resume
+        )
+
+        if resume and "unknown field `resume_vm`" in response.text:
```

Comment:
nit: can this be replaced with an actual version check ?

Reply:
I think the framework is more robust this way. Versions are binary/branch specific while this code should work across the board.

Ideally, we want to drop these workarounds and do https://github.com/firecracker-microvm/firecracker/issues/2315

Reply:
Makes sense, let's prioritize #2315  in v0.25.